gamemodes/terrortown/entities/entities/ttt_beacon.lua

```diff
@@ -0,0 +1,222 @@
+---
+-- @class ENT
+-- @section ttt_beacon
+
+if SERVER then
+	AddCSLuaFile()
+end
+
+if CLIENT then
+	ENT.Icon = "vgui/ttt/icon_beacon"
+	ENT.PrintName = "Beacon"
+end
+
+ENT.Type = "anim"
+ENT.Model = Model("models/props_lab/reciever01b.mdl")
+ENT.CanHavePrints = true
+ENT.CanUseKey = true
+
+local beaconDetectionRange = 135
+
+local soundZap = Sound("npc/assassin/ball_zap1.wav")
+local soundBeep = Sound("weapons/c4/cc4_beep1.wav")
+
+---
+-- @realm shared
+function ENT:Initialize()
+	self:SetModel(self.Model)
+
+	if SERVER then
+		self:PhysicsInit(SOLID_VPHYSICS)
+	end
+
+	self:SetMoveType(MOVETYPE_VPHYSICS)
+	self:SetSolid(SOLID_BBOX)
+	self:SetCollisionGroup(COLLISION_GROUP_WEAPON)
+
+	if SERVER then
+		self:SetMaxHealth(100)
+	end
+	self:SetHealth(100)
+
+	if SERVER then
+		self:SetUseType(SIMPLE_USE)
+		self:NextThink(CurTime() + 1)
+	end
+end
+
+---
+-- @param Entity activator
+-- @realm shared
+function ENT:UseOverride(activator)
+	if not IsValid(activator) or self:GetOwner() ~= activator then return end
+
+	local wep = activator:GetWeapon("weapon_ttt_beacon")
+
+	if IsValid(wep) then
+		local pickup = wep:PickupBeacon()
+
+		if not pickup then return end
+	else
+		wep = activator:GiveEquipmentWeapon("weapon_ttt_beacon")
```

Comment:
We should make sure that this then only has 1 ammo in it, otherwise i would be able to pickup the last beacon and reset my ammo to place infinite beacons.

Reply:
We can reduce this to one if you like, yes. But I don't get your exact problem here.

You have three beacons. every time you drop one, the ammo is reduced by one. If you drop all three, you have non left. If you pick up one again, then you have one back in your inventory. You can only plant three, not an infinite ammunt.

The marked line gives you a beacon with 1 ammo. It is only set to three in the `WasBought` callback: https://github.com/TTT-2/TTT2/blob/0cb5472df06884a48f005a7e0978a4d8bb01c91c/gamemodes/terrortown/entities/weapons/weapon_ttt_beacon.lua#L105-L107